### PR TITLE
Skip copying jfr recoding into a temporary file

### PR DESCRIPTION
**Pull request by laurit**

### CODE REVIEW

File: profiler/README.md [hunk 1]
```diff
@@ -105,14 +105,6 @@ The agent logs the profiling configuration at `INFO` during startup. You can gre
...
 ```
 
-### What about this escape hatch?
-
-If the escape hatch becomes active, it will log with `com.splunk.opentelemetry.profiler.RecordingEscapeHatch`
-(you can grep for this in the logs). You may also look for `"** THIS WILL RESULT IN LOSS OF PROFILING DATA **"`
-as a big hint that things are not well.
-
-You may need to free up some disk space and/or give the JVM more resources.
-
```
Comment (theletterf) [inline profiler/README.md:115]: @laurit @breedx-splk So this is gone? If that's the case, I'll remove it from the official troubleshooting.

**Review by mateuszrzeszutek — APPROVED** [review]
Nice!

**Comment by breedx-splk** [comment]
Oh I think I had missed that we were doing that copy in the first place. Seems like a good idea to prevent that if we can.

I'm not sure why this change also removes support for the escape hatch. Shouldn't we still use that? We don't want to fill up the disk.

**Comment by laurit** [comment]
> I'm not sure why this change also removes support for the escape hatch. Shouldn't we still use that? We don't want to fill up the disk.

As we are not copying the recording to a file there is no need to check for free space. As for backlogged check I suspect it didn't work as these files were processed one at a time, previous one was deleted before the next was written do disk.

**Comment by breedx-splk** [comment]
> > I'm not sure why this change also removes support for the escape hatch. Shouldn't we still use that? We don't want to fill up the disk.
> 
> As we are not copying the recording to a file there is no need to check for free space. As for backlogged check I suspect it didn't work as these files were processed one at a time, previous one was deleted before the next was written do disk.

Ah ok, I hadn't realized that we're not writing to disk at ALL now in the normal flow. I had assumed that the recordings were still to disk, but I do see that we `setToDisk(false)` in `start()`. Seems good to me.

I suppose there is chance that we could fill up the disk with the keepfiles setting, but I'm also ok with that since it's only really for development/testing purposes anyway.

**Comment by theletterf** [comment]
Roger! The troubleshooting section on the escape hatch has been removed now.

**Pull request merged**